src/Units/UnitModel.java

```diff
@@ -147,9 +147,43 @@ public boolean canRepairOn(Location locus)
     return new ArrayList<GameAction>();
   }
 
+  /**
+   * @return True if this UnitModel has at least one weapon with a minimum range of 1.
+   */
+  public boolean hasDirectFireWeapon()
```

Comment:
I feel like this is maybe of questionable utility. What you're really looking for, I feel, is units that can fire after moving.
All having range 1 means is the ability to counterattack.

Reply:
Why is that what I really want? Muriel calculates cost-efficiency by using damage ratio and factoring in cost. Damage ratio is either zero or infinity for indirect-fire interactions, so I'm just ignoring those for now so they don't influence my unit-purchase calculations.

Reply:
I feel like that's a Muriel-specific calculation, then.
Whether a unit can shoot at 1 range is much less important than whether it can fire after moving... and it's a 100% correlation at this point.

Reply:
Well, we can certainly add utility functions for `canFireAfterMoving` and `hasRangedWeapon` and `canCapture` and whatever silly else we think of. I chose `hasDirectFireWeapon` because that is precisely what I cared about. The fact that it's correlated with another trait doesn't really matter.